sherpa/sim/__init__.py

```diff
@@ -1004,3 +1008,78 @@ def call(self, niter, seed=None, rng=None,
             result[name] = pars[name]
 
         return result
+
+
+# The return value is InterferenceData, but this is not imported on the module
+# level, so we can't use it in the type hint.
+def mcmc_to_arviz(mcmc : MCMC, fit: Fit, list_of_draws : list[tuple],
+                  ):
+    """Convert the MCMC results to an ArviZ InferenceData object.
+
+    ArviZ is a Python package for exploratory analysis of Bayesian models and
+    provides a range of tools for visualizing and analyzing MCMC results beyond
+    what is implemented in Sherpa itself.
+
+    This function requires the `ArviZ <https://python.arviz.org>`_ package to be installed.
+
+    Parameters
+    ----------
+    mcmc : MCMC
+        The MCMC object used to run the chain.
+    fit : Fit
+        The Fit object used to fit the data.
+    list_of_draws : list of tuples
+        A list of tuples, each containing three elements as returned from
+        `MCMC.get_draws`: the statistic, the acceptance rate, and the
+        parameter array. Each tuple corresponds to a single chain run for the
+        same model and MCMC object.
+
+    Returns
+    -------
+    inference_data: arviz.InferenceData
+        An ArviZ InferenceData object containing the posterior and log likelihood data.
+    """
+
+    from arviz.data.base import dict_to_dataset
+    from arviz.data.inference_data import InferenceData
+
+    pnames = [p.fullname for p in fit.model.get_thawed_pars()]
+
+    for i, draw in enumerate(list_of_draws):
+        if len(draw) != 3:
+            raise ValueError("For each chain in the list_of_draws, there must be three elements: " +
+                             "the statistic, the acceptance rate, and the parameter array.")
+        if draw[2].shape[0] != len(pnames):
+            raise ValueError(f"Chain {i} contains {draw[2].shape[0]} dimensions, but the model has {len(pnames)} thawed parameters.")
+        if len(draw[0]) != draw[2].shape[1]:
+            raise ValueError(f"Chain {i} contains {draw[0].shape[0]} steps for the static, but parameter array has {draw[2].shape[1]} steps.")
+        if len(list_of_draws[0][0]) != len(draw[0]):
+            raise ValueError("All chains must have the same number of steps. " +
+                f"Chain {i} contains {draw[0].shape[0]} steps, but the first chain has {len(list_of_draws[0][0])} steps.")
+
```

Comment:
It would help here if we had a way to structure the MCMC data into something more structured than a bunch of tuples, but that is a future-us issue!